orchestrator/src/compression/blob.rs

```
@@ -0,0 +1,406 @@
+        .map(|i| {
+            let mut result = BigUint::zero();
+            let mut xi_pow_j = ONE.clone(); // Initialize to xs[i]**0
+
+            for j in 0..arr.len() {
+                let term = (&arr[j] * &xi_pow_j) % p;
+                result = (result + term) % p;
+                xi_pow_j = (&xi_pow_j * &xs[i]) % p; // Update power for next iteration
+            }
+
+            result
+        })
+        .collect()
+}
+
+/// Generate evaluation points for NTT/IFFT operations
+///
+/// # Arguments
+/// * `size` - Size of the data (default: BLOB_LEN)
+///
+/// # Returns
+/// A tuple of (evaluation points, modulus)
+pub fn generate_evaluation_points(size: Option<usize>) -> (Vec<BigUint>, BigUint) {
+    let blob_len = size.unwrap_or(BLOB_LEN);
+
+    // Generate evaluation points
+    let xs: Vec<BigUint> = (0..blob_len)
+        .map(|i| {
+            let bin = format!("{:012b}", i);
+            let bin_rev = bin.chars().rev().collect::<String>();
+            GENERATOR.modpow(&BigUint::from_str_radix(&bin_rev, 2).unwrap(), &BLS_MODULUS)
+        })
+        .collect();
+
+    (xs, BLS_MODULUS.clone())
+}
+
+/// Process a vector of BigUint values for blob data transformation using NTT
+///
+/// # Arguments
+/// * `data` - Vector of BigUint values to process
+/// * `size` - Optional size parameter to override BLOB_LEN (default: BLOB_LEN)
+///
+/// # Returns
+/// Transformed vector for blob data
+pub fn process_for_blob(data: Vec<BigUint>, size: Option<usize>) -> Vec<BigUint> {
+    let blob_len = size.unwrap_or(BLOB_LEN);
+
+    // Get evaluation points
+    let (xs, p) = generate_evaluation_points(Some(blob_len));
+
+    // Ensure data is of correct length
+    let mut data_padded = data;
+    if data_padded.len() < blob_len {
+        // Pad with zeros if needed
+        data_padded.resize(blob_len, BigUint::zero());
+    } else if data_padded.len() > blob_len {
+        // Truncate if too long
+        data_padded.truncate(blob_len);
+    }
```

here let's not truncate the extra data, because ideally we should be sending only vec of size 4096, this might cause some silent errors

---

removed this function. was not being used